monitoring/mock_uss/ridsp/user_notifications.py

```diff
@@ -36,28 +36,57 @@ def create_notifications_if_needed(self, record: "database.TestRecord"):
         for notif in check_and_generate_slow_update_notification(record.flights):
             self.record_notification("Insufficient update rate", notif)
 
-        for notif in check_and_generate_missing_fields_notifications(record.flights):
-            self.record_notification(notif)
+        for notif_date, notif_str in check_and_generate_missing_fields_notifications(
+            record.flights
+        ):
+            self.record_notification(message=notif_str, observed_at=notif_date)
 
 
 def check_and_generate_missing_fields_notifications(
     injected_flights: List[TestFlight],
-) -> List[str]:
+) -> List[Tuple[datetime.datetime, str]]:
+
     missing_fields_notifications = []
 
     for flight in injected_flights:
+
+        default_timestamp = None
+
+        f_start, _ = flight.get_span()
+        if f_start:  # get_span may fail to find a start
+            default_timestamp = arrow.get(f_start)
+
+        if not default_timestamp:  # If we didn't find anything, default to now
+            default_timestamp = arrow.utcnow()
```

Comment:
I'm a bit confused: isn't this strictly equivalent? Or am I missing something?
(c.f. https://github.com/interuss/monitoring/pull/935/files#r1950612015)
```suggestion

        default_timestamp = arrow.utcnow()

        f_start, _ = flight.get_span()
        if f_start:  # get_span may fail to find a start
            default_timestamp = arrow.get(f_start)
```

Reply:
haaa yes, ok, I was thinking the other way around, that equivalent yes ^^'